lib/core.js

```diff
@@ -27,26 +26,62 @@ import { promiseRetry, bind, uuid } from './util';
  * @param {CoreConfig} cfg
  */
 export function Core(cfg) {
-  this.Promise = cfg.Promise || window.Promise;
+  this.Promise = cfg.Promise;
 
   // API connectivity.
   this.key = cfg.key;
   this.api = cfg.api || apiFromKey(this.key);
   this.api = this.api[0] + this.api.substr(1).replace(/^\/+|\/$/g, '');
   this.sendRetryMs = cfg.sendRetryMs || 150;
-  this.idExpiryDays = cfg.idExpiryDays || 365;
-
-  // Device IDs.
+  this.cookie = cfg.cookie || 'ravelinDeviceId';
+  this.cookieExpiryDays = cfg.cookieExpiryDays || 365;
   this.cookies = new CookieJar({
     domain: cfg.cookieDomain
   });
-  this.sync();
-  if (cfg.init !== false) {
-    this.attach(cfg.syncMs || 2000);
+
+  if (cfg.id) {
+    this._id = this._idBackup(cfg.id);
```

Comment:
Do you need to run `sync()` or `attach()` after resolving the device id? Not sure how that part works so I'm probably missing something.

Reply:
If the user specifies their own ID I don't want to to risk tinkering with it by running sync(). We'll just assume they've taken care of it.